react-state/my-app/src/ImageContainer.tsx

```diff
@@ -0,0 +1,30 @@
+import './ImageContainer.css';
+import { useState } from 'react';
+
+type Props = {
+  imageSrc: string[];
+};
+
+export function ImageContainer({ imageSrc }: Props) {
+  const [index, setIndex] = useState<number>(0);
```

Comment:
You only need the generic if you don't plan on assigning your state variable a default value. TypeScript is able to infer the type from the default value.

If you did `useState<number>()`, we'd need the generic for number because TypeScript doesn't know what's getting passed into `useState`, so we need to tell it